src/mqt/qecc/ft_stateprep/state_prep.py

```diff
@@ -1124,6 +1208,8 @@ def _symbolic_vector_add(
 
 def _odd_overlap(v_sym: npt.NDArray[z3.BoolRef | bool], v_con: npt.NDArray[np.int8]) -> z3.BoolRef:
     """Return True if the overlap of symbolic vector with constant vector is odd."""
+    if np.array_equal(v_con, np.zeros(len(v_con), dtype=np.int8)):
+        return False
```

Comment:
Fixed it, not sure why mypy did not complain about it before but should be fine now.

Reply:
Yeah. I was also kind of surprised by that. It was only PyCharm that was complaining.
Might be worth investigating at some point whether mypy is really set up properly in QECC. For now this can be resolved.